python/ray/data/_internal/logical/optimizers.py

```diff
@@ -20,35 +16,44 @@
     EliminateBuildOutputBlocks,
 )
 from ray.data._internal.planner.planner import Planner
+from ray.util.annotations import DeveloperAPI
 
-DEFAULT_LOGICAL_RULES = [
+_LOGICAL_RULES = [
     ReorderRandomizeBlocksRule,
 ]
 
-DEFAULT_PHYSICAL_RULES = [
+_PHYSICAL_RULES = [
     InheritTargetMaxBlockSizeRule,
     SetReadParallelismRule,
     OperatorFusionRule,
     EliminateBuildOutputBlocks,
 ]
 
 
+@DeveloperAPI
+def register_logical_rule(cls: Type[Rule]):
+    _LOGICAL_RULES.append(cls)
```

Comment:
Generally, modifying constants is not ideal. one alternative could be to create a class which manages the rules, something like:
```
class OptimizationRuleset:
    def register_logical_rules():

    def get_logical_rules():

    def register_physical_rules():

    def get_physical_rules():

```

Reply:
discussed offline, the additional complexity of adding a separate class is not worth it, since these are internal values which the vast majority of users should not need to access or modify.